src/Foundatio/Caching/InMemoryCacheClient.cs

```diff
@@ -797,19 +820,20 @@ public Task<bool> ExistsAsync(string key)
         if (String.IsNullOrEmpty(key))
             throw new ArgumentNullException(nameof(key), "Key cannot be null or empty");
 
-        if (!_memory.TryGetValue(key, out var value) || value.ExpiresAt == DateTime.MaxValue)
+        if (!_memory.TryGetValue(key, out var existingEntry) || existingEntry.ExpiresAt == DateTime.MaxValue)
         {
             Interlocked.Increment(ref _misses);
             return Task.FromResult<TimeSpan?>(null);
         }
 
-        Interlocked.Increment(ref _hits);
-        if (value.ExpiresAt >= SystemClock.UtcNow)
-            return Task.FromResult<TimeSpan?>(value.ExpiresAt.Subtract(SystemClock.UtcNow));
-
-        RemoveExpiredKey(key);
```

Comment:
Just let maintenance remove the item to cut down on the number concurrent updates.